ui/tests/e2e/pages/ModelsPage.js

```diff
@@ -0,0 +1,222 @@
+    await this.searchInput.click();
+    await this.searchInput.fill(modelDisplayName);
+    await this.page.getByText(modelDisplayName).click();
+  }
+
+  async exportModel() {
+    const downloadPromise = this.page.waitForEvent('download');
+    await this.exportModelButton.click();
+    const download = await downloadPromise;
+    expect(download).toBeDefined();
+    return download;
+  }
+
+  async changeModelStatus() {
+    await this.enabledCombobox.click();
+    await this.ignoredOption.click();
+    await expect(this.ignoredCombobox).toBeVisible();
+  }
+
+  async selectFileImport() {
+    await this.fileImportHeading.click();
+  }
+
+  async selectUrlImport() {
+    await this.urlImportHeading.click();
+  }
+
+  async selectCsvImport() {
+    await this.csvImportHeading.click();
+  }
+
+  async uploadFile(filePath) {
+    await this.fileInput.setInputFiles(filePath);
+  }
+
+  async fillUrlImport(url) {
+    await this.urlImportTextbox.click();
+    await this.urlImportTextbox.fill(url);
+  }
+
+  async proceedWithImport() {
+    await this.nextButtonGeneric.click();
+  }
+
+  async finishImport() {
+    await this.finishButtonGeneric.click();
+  }
+
+  async generateFromCsv() {
+    await this.generateButtonGeneric.click();
+  }
+
+  // Complete workflow methods
+  async createModelWorkflow(modelData) {
+    await this.clickCreateModel();
+    await this.fillModelDetails(modelData.MODEL_NAME, modelData.MODEL_DISPLAY_NAME);
+    await this.proceedThroughCategoryStep();
+    await this.proceedThroughStylingStep();
+    await this.configureGitHubSource(modelData.MODEL_URL);
+    await this.enableVisualAnnotationAndGenerate();
+    await this.verifyModelImported(modelData.MODEL_NAME);
+    await this.finishModelCreation();
+  }
+
+  async importModelViaFile(modelData) {
+    await this.clickImportModel();
+    await this.selectFileImport();
+    await this.uploadFile(modelData.MODEL_FILE_IMPORT);
+    await this.proceedWithImport();
+    await this.verifyModelImported(modelData.MODEL_NAME);
+    await this.finishImport();
+  }
+
+  async importModelViaUrl(modelData) {
+    await this.clickImportModel();
+    await this.selectUrlImport();
+    await this.fillUrlImport(modelData.MODEL_URL_IMPORT);
+    await this.proceedWithImport();
+    await this.verifyModelImported(modelData.MODEL_NAME);
+    await this.finishImport();
+  }
+
+  async importModelViaCsv(csvData) {
+    await this.clickImportModel();
+    await this.selectCsvImport();
+    await this.proceedWithImport();
+
+    // Upload models CSV
+    await this.uploadFile(csvData.Models);
+    await this.proceedWithImport();
+
+    // Upload components CSV
+    await this.uploadFile(csvData.Components);
+    await this.proceedWithImport();
+
+    // Upload relationships CSV
+    await this.uploadFile(csvData.Relationships);
+    await this.generateFromCsv();
+
+    await this.verifyModelImported(csvData.Model_Name);
+    await this.finishImport();
+  }
```

Comment:
These large multi-step funciton would be good to have some expect as the test is progressing in the spec itself.